sechub-shared-kernel/src/test/java/com/mercedesbenz/sechub/sharedkernel/TimeCalculationServiceTest.java

```diff
@@ -0,0 +1,45 @@
+package com.mercedesbenz.sechub.sharedkernel;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullSource;
+
+class TimeCalculationServiceTest {
+
+    private TimeCalculationService serviceToTest;
+    private SystemTimeProvider systemTime;
+    private LocalDateTime date_2021_02_01_time_08_15;
+
+    @BeforeEach
+    void beforeEach() {
+        serviceToTest = new TimeCalculationService();
+
+        systemTime = mock(SystemTimeProvider.class);
+        date_2021_02_01_time_08_15 = LocalDateTime.of(2021, 2, 1, 8, 15);
+
+        serviceToTest.systemTime = systemTime;
+    }
+
+    @NullSource
+    @CsvSource({ "0", "1", "90", "365" })
+    @ParameterizedTest
+    void now_minus_days_calulation_works(Long days) {
```

Comment:
What happens if one deducts so many days, that the result will be negative? Do I get an exception?

Reply:
The time calculation service does not know that negative values are not acceptable. it will add them.

I added tests to ensure the behaviour.

I changed the implementation inside configuration update mechanism and also inside the execution + added tests for this.
So
- it cannot be configured with negative values by admins
- even when somebody changes value in database, the execution will just log an error but not accept it any longer.

Thanx for finding that.